Engine/src/mini/input/Mouse.java

```diff
@@ -26,6 +29,16 @@
      */
     public static void mouseAction(int button, int action, int mods) {
         actions.put(button, action);
+
+        MouseButton btn = MouseButton.values()[button];
+        for(MouseListener listener : listeners) {
+            if (action == 1) {
+                listener.OnClick(btn, lastXPos, lastYPos);
+            } else if (action == 0) {
+                listener.OnRelease(btn, lastXPos, lastYPos);
+            }
+        }
```

Comment:
Pas her på med hvordan du optimere dine kald. Din forløkke kigger på action for hver eneste listener. Hvis du istedet havde ude omkring, ville det ikke kræve et if check hver gang. Måske er compileren god nok til at optimere det væk. But a small pointer